### PR TITLE
Feature/multi file inference

### PR DESCRIPTION
**Pull request by mosheman5**
Now we can:
- perform inference either on a single file or multiple files
- we get multiple classes merged to one raven file in case we have classification model
- a script to merge multiple raven files to a single big raven file
- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -

### CODE REVIEW

**Review by shaing10 — APPROVED** [review]
Let's merge it can make our lives much easier :)

File: soundbay/inference.py [hunk 1]
```diff
@@ -180,64 +153,62 @@ def infer_single_file(
...
+                file_raven_lists.append(
+                                       inference_csv_to_raven(results_df=df,
+                                                              num_classes=predict_prob.shape[1],
+                                                              seq_len=dataset_args['seq_length'],
+                                                              selected_class=label_names[i],
+                                                              threshold=threshold,
+                                                              class_name=label_names[i],
+                                                              max_freq=dataset_args['sample_rate'] // 2)
+                                   )
+            whole_file_df = pd.concat(file_raven_lists, axis=0).sort_values('Begin Time (s)')
```
Comment (shaing10) [inline soundbay/inference.py:185]: suggestion: first by file, and then by begin_time?

File: soundbay/inference.py [hunk 2]
```diff
@@ -180,64 +153,62 @@ def infer_single_file(
...
+            whole_file_df['Selection'] = np.arange(1, len(whole_file_df) + 1)
+            all_raven_list.append((file, whole_file_df))
 
     #save file
     dataset_name = Path(test_dataset.metadata_path).stem
     filename = f"Inference_results-{datetime.datetime.now().strftime('%Y-%m-%d_%H-%M-%S')}-{model_name}-{dataset_name}.csv"
     output_file = output_path / filename
-    pd.concat(all_channel_list, axis=0).to_csv(index=False, path_or_buf=output_file)
+    concat_dataset.to_csv(index=False, path_or_buf=output_file)
```
Comment (shaing10) [inline soundbay/inference.py:193]: before 193, suggestion:
`concat_dataset = concat_dataset.sort_values(by=['filename', 'begin_time'])`

File: soundbay/inference.py (hunk 1 above)
Comment (mosheman5) [reply soundbay/inference.py:185]: same as next suggestion, resolving

File: scripts/merge_multiple_ravens_to_one_file.py [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+    # iterate over the raven files
+    seconds_offset = 0
+    entries_offset = 0
+    for entry in tqdm(adapted_files_list):
+        # read the raven file
+        df = pd.read_csv(entry["raven_file"], sep="\t")
+        # add the offset to the begin and end time
+        df['Begin Time (s)'] += seconds_offset
+        df['End Time (s)'] += seconds_offset
+        df['Selection'] += entries_offset
```
[the remark this answers is not in the document]
Comment (mosheman5) [reply scripts/merge_multiple_ravens_to_one_file.py:60]: We already have an additional column with the original source file (`df['Begin File']`)
I've also created an index file to make it more explicit

File: scripts/merge_multiple_ravens_to_one_file.py [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+    args = make_parser().parse_args()
+    raven_folder = Path(args.input_raven_folder)
+    audio_folder = Path(args.input_audio_folder)
+    output_path = Path(args.output_path)
+    # get the list of raven files
+    raven_files = list(raven_folder.glob('*.txt'))
+    # get the list of audio files
+    audio_files = list(audio_folder.glob('*.wav'))
+    # sort the audio files by name, should be the order by start time as well
+    audio_files = sorted(audio_files)
```
[the remark this answers is not in the document]
Comment (mosheman5) [reply scripts/merge_multiple_ravens_to_one_file.py:38]: Not gonna implement it now, if needed in the future we can have a user defined key ordering

**Pull request merged**